design-proposals/app-orch-helm-to-dp.md

```diff
@@ -0,0 +1,149 @@
+imported, the user may make use of existing pages in the GUI to add profiles, parameter templates, and
+other features to their `Deployment Package`. These workflows are already documented in the
+Orchestrator User Guide. The `Helm Chart Import` provides a starting point to this optional value
+add.
+
+### Export
+
+The final step in the workflow is to export any work that the user has done, so that it may be shared
+in other orchestrators. The export will be in a `.tar.gz` file that contains the `yaml` representation
+of the `Deployment Package`. Exporting this as a `.tar.gz` file is convenient to do in the user's browser
+as it allows a single file to be downloaded.
+
+As with import, there are two options
+
+- `Command Line Tool`. The CLI could be used to reach out to the Orchestrator, download the necessary
+  parts of the Deployment Package, and emit them locally as either a `.tar.gz` file or as a set of
+  individual yaml files.
+
+- `GUI Export`. A button could be added to the GUI that triggers the export by reaching out to the
+  backend to generate a `.tar.gz` file, which is then downloaded via the users browser.
+
+These two options are not mutually exclusive -- we can do both.
+
+## Rationale
+
+We considered both CLI and GUI approaches. CLI has the advantage that it's usable by people who do not
+have access to an orchestrator, and can be used to craft `Deployment Package` that could be handed
+off to another team for evaluation and use. GUI has the advantage requires a lower knowledge set and
+facilitates a quick path toward bringing in Helm Charts directly.
+
+The GUI is the more user-facing and more visible component, and we propose prioritizing GUI over CLI,
+but also that we should strive to do both.
+
+## Affected components and Teams
+
+- Application Orchestration Application Catalog.
+
+- GUI.
+
+- CLI.
+
+## Implementation plan
+
+- App Orch. Implement Helm-to-DP command line tool, generating yaml files in a local directory.
+
+- App Orch. Extend Catalog Service Import to allow `.tar.gz` files to be imported.
+
+- App Orch. Implement Catalog Service API endpoint to invoke Helm-to-DP based on REST API. This
+  API will block while the Helm Chart is being fetched, and return an error if the fetch is
+  unsuccessful or times out.
+
+- GUI. Implement UI page for accepting OCI URL, Username/Password, and initial values. It will call
+  the REST API endpoint in the Catalog Service and return a green success box if successful, or
+  the error if unsuccessful.
+
+- App Orch. Implement Catalog Service API endpoint to export DP package and generate a tarball for
+  download.
+
+- GUI. Implement UI button or link for exporting deployment package.
+
+## Open issues (if applicable) / Feature Requests
+
+- Automatically adding the chart's values.yaml as the initial profile if the user did not provide their
+  own values.yaml when importing. This is a feature request from a VBU user. The only disadvantage is
```

Comment:
May be we should say this is requirement from internal customers.

Reply:
updated.